src/layout/core/borders/borders.rs

```diff
@@ -16,6 +25,10 @@ pub struct Borders {
     pub title: String,
     /// The surface that contains the bytes we give to wlc to draw.
     surface: ImageSurface,
+    /// Children titles to be used tabbed/stacked layouts
+    pub children: Option<Children>,
+    /// Layout of the container, only used if the border is from a container
+    pub layout: Option<Layout>,
```

Comment:
Hmm, with all of these optional fields it might be prudent to split this up into a separate type of borders struct (e.g `ContainerBorders` or `ViewBorders`).

I don't know how annoying that would be though, especially as they would probably share a _lot_ of code (and there's no real notion of inheritance in Rust like you see in more traditional OO languages obviously).

You could probably make it work with an enum though, where only some of the functions split based on the type and other wise just do the same thing. It would increase the size of the borders struct slightly because there needs to be a tag, so you wouldn't get the size benifits of splitting them up but it would be easier to follow the logic and harder to mess up (and create another layer of abstraction, e.g borders for containers vs borders for views instead of the current setup where it's just borders that are sometimes used for one or the other).

If this is a lot of work, we can push this to a future patch/issue.

Reply:
I feel the same, at this point there are too many differences between the borders of views vs containers and even the borders of each layout, because ideally (going i3) the other layouts shouldn't even draw borders (vertical/horizontal), but I do think this should be part of a future redesign.